snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -90,6 +97,26 @@ def __init__(self, *, plugin, part_properties, project_options,
 
         self._migrate_state_file()
 
+    def pull_state(self) -> states.PullState:
```

Comment:
these should have a `get` in front

Reply:
maybe this really merits a mixin as well? Thoughts?

Reply:
Oh right-- I'll fix the lack of "get". Regarding the mixin, I feel it's only worth it if we want the functionality mixed into another class as well. I don't think we do... do we?

Reply:
Fixed.